pulpcore/content/handler.py

```diff
@@ -1335,6 +1335,12 @@ async def finalize():
                 "on-demand-downloading/#on-demand-and-streamed-limitations>"
             )
 
+        # manually close the DownloadFactory's (HTTP-)session, the next artifact-download will
+        # create a new DownloadFactory anyway because it will use a new remote-object.
+        # Leaving it open also left open file-descriptors to /dev/urandom. Most likely these FDs are
+        # held by the underlying SSL library.
+        await remote.download_factory._session.close()
```

Comment:
Does this give the same result?
```suggestion
        await downloader.session.close()
```

Reply:
@gerrod3 Seems to work just as well.